SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -537,13 +539,10 @@ void UGlobalStateManager::BecomeAuthoritativeOverAllActors()
 	for (TActorIterator<AActor> It(NetDriver->World); It; ++It)
 	{
 		AActor* Actor = *It;
-		if (Actor != nullptr && !Actor->IsPendingKill())
+		if (Actor != nullptr && !Actor->IsPendingKill() && USpatialWorker::CanHaveAuthority(Actor) && Actor->GetIsReplicated())
```

Comment:
This should also check for singleton actors and only apply if offloading is enabled...